uclahs-cds/lintr
# Add single quotes, dotted.mixed.case to v3.0.1

## Migration Notes

version: r-lib v3.0.1.9000 (development) as of 2022-09-16

<h4> Summary of Changes:</h4>
<blockquote>
<ul>
<li><p>set modifications/params on default linters to use in  <code>.lintr</code> (main config)</p></li>
<li><p>add dotted.mixed.case </p></li>
<li><p>change single quote linting behavior</p></li>
<li><p>remove unnecessary r-lib GitHub Actions </p></li>
</ul>
</blockquote>


--- 
### [Detailed Changes]

#### Config changes

- Update linters in `.lintr` (main config file)
- Update defaults used in test action `tests/testthat/dummy_packages/package/lintr_test_config`


#### Dotted.Mixed.Case

Updates required:
- `R/object_name_lintr.R`
   - add `dotted.mixed.case`(different from `dotted.case`) 
   - ```
     "dotted.mixed.case" = rex(start, maybe("."), one_or_more(alphanum), zero_or_more(dot, one_or_more(alphanum)), end),
     ```
Test case files:
- `test-object_name_lintr.R`
   - update test case matrix for `dotted.mixed.case` (same values as `dotted.case` but allow "Uppercase")
   - update message (the one with dotted.case) to include `dotted.mixed.case`
       - "Variable and function name style should be camelCase, dotted.case or dotted.mixed.case." 
       - Note: no comma before "or". See logic in object_name_lintr.

#### Single Quotes (lab version)

Updates required for using lab's single_quote:
- `R/single_quotes_lintr.R`
    - replace text to say single-quote instead of double-quote
    - change `squote_regex`
  ```
  squote_regex <- rex(start, zero_or_one(character_class("rR")), double_quote, any_non_single_quotes, double_quote, end)
  ```
For all of the test case files:
- replace every double quote `"` with single quote `'` and vice versa (only if it's text in a test)
- replace message text to say `single-quote` instead of `double-quote`

Test case files:
- `inst/example/bad.R` 
- `tests/test-single_quotes_lintr`
- `tests/testthat/dummy_packages/clean/R/clean_generics.R`
- `tests/testthat/dummy_packages/clean/R/default_lintr_testcode.R`
- `tests/testthat/test-knitr_formats.R`

#### GitHub Actions:
- remove GitPages deployment 
      - `pkgdown` files
- disable R linting "Error on Lint Failure" on the Git Action for this package <br>&emsp;___Note:___ _Will cause fails from `r-lib` not using `single quotes` or `dotted.mixed.case`_
